src/common/timings.cc

```diff
@@ -73,53 +74,100 @@ bool TimingsDatabase::load()
     {
       i.deciles.push_back(atoi(fields[idx++].c_str()));
     }
-    instances.insert(std::make_pair(name, i));
+    instances.emplace_back(name, i);
   }
   fclose(f);
   return true;
 }
 
-bool TimingsDatabase::save()
+bool TimingsDatabase::save(const bool save_current_time /*=true*/)
 {
-  if (filename.empty())
+  if (filename.empty() || instances.empty())
     return true;
 
-  FILE *f = fopen(filename.c_str(), "w");
+  FILE *f = fopen(filename.c_str(), "a");  // append
   if (!f)
   {
     MERROR("Failed to write to file " << filename << ": " << strerror(errno));
     return false;
   }
-  for (const auto &i: instances)
+
+  if (save_current_time)
   {
-    fprintf(f, "%s", i.first.c_str());
-    fprintf(f, "\t%lu", (unsigned long)i.second.t);
-    fprintf(f, " %zu", i.second.npoints);
-    fprintf(f, " %f", i.second.min);
-    fprintf(f, " %f", i.second.max);
-    fprintf(f, " %f", i.second.mean);
-    fprintf(f, " %f", i.second.median);
-    fprintf(f, " %f", i.second.stddev);
-    fprintf(f, " %f", i.second.npskew);
-    for (uint64_t v: i.second.deciles)
-      fprintf(f, " %lu", (unsigned long)v);
+    // save current time in readable format (UTC)
+    std::time_t sys_time{std::time(nullptr)};
+    std::tm *utc_time = std::gmtime(&sys_time);    //GMT is equivalent to UTC
+
+    // format: year-month-day : hour:minute:second
+    std::string current_time{};
+    if (utc_time && sys_time != (std::time_t)(-1))
+    {
+        current_time += std::to_string(utc_time->tm_year + 1900) + '-';
+        current_time += (std::to_string(utc_time->tm_mon + 1).size() == 1 ? std::string{'0'} : std::string{}) +
+          std::to_string(utc_time->tm_mon + 1) + '-';
+        current_time += (std::to_string(utc_time->tm_mday).size() == 1 ? std::string{'0'} : std::string{}) +
+          std::to_string(utc_time->tm_mday) + " : ";
+        current_time += (std::to_string(utc_time->tm_hour).size() == 1 ? std::string{'0'} : std::string{}) +
+          std::to_string(utc_time->tm_hour) + ':';
+        current_time += (std::to_string(utc_time->tm_min).size() == 1 ? std::string{'0'} : std::string{}) +
+          std::to_string(utc_time->tm_min) + ':';
+        current_time += (std::to_string(utc_time->tm_sec).size() == 1 ? std::string{'0'} : std::string{}) +
+          std::to_string(utc_time->tm_sec);
+    }
+    else
+    {
+        current_time += "TIME_ERROR_";
+    }
+    fputc('\n', f);  // add an extra line before each 'print time'
+    fprintf(f, "%s", current_time.c_str());
     fputc('\n', f);
   }
+
+  for (const auto &i: instances)
+  {
+    fprintf(f, "%s,", i.first.c_str());
+
+    if (i.second.npoints > 0)
+    {
+      fprintf(f, "%lu,", (unsigned long)i.second.t);
```

Comment:
Its _probably_ more efficient to collapse these into one `fprintf` call, but that may mean worse formatting of the code.